include/xtensor/xslice.hpp

```diff
@@ -656,8 +629,39 @@ namespace xt
             return xall<std::ptrdiff_t>(static_cast<std::ptrdiff_t>(size));
         }
 
+        A start() const { return m_start; }
+        B stop()  const { return m_stop;  }
+        C step()  const { return m_step;  }
+
     private:
 
+        static auto normalize(const std::ptrdiff_t val,const std::size_t ssize)
```

Comment:
No need to put const here since the value are passed by copy

Reply:
Well I usually like to put the const as a marker for people reading  the code. In this case, the const on val (and similarly on start/stop in the next function) are wrong as those values are modified...
And this broke xtensor-blas... But I'll remove them (I mean the ones that are not buggy) if you prefer...